featurebyte/service/feature_readiness.py

```diff
@@ -301,6 +303,7 @@ async def _validate_readiness_transition(
                 "Please delete the feature instead if it is no longer needed."
             )
 
+    @async_retry(max_retries=3, exceptions=(OperationFailure,), backoff_factor=2)
```

Comment:
nit: can potentially skip the `max_retries` and `backoff_factor` parameters here since they're the same as the default